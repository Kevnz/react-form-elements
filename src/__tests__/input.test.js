import React from 'react'
import { render, fireEvent, cleanup } from '@testing-library/react'

import Input from '../input'

describe('Input component', () => {
  afterEach(() => {
    cleanup()
  })
  describe('The Rendered HTML', () => {
    it('should render input element', () => {
      const { container } = render(<Input initialValue="" />)

      expect(container.firstChild).toMatchSnapshot()
    })
    it('should have a placeholder when passed', () => {
      const { container } = render(
        <Input initialValue="" placeholder="textbox placeholder" />
      )

      expect(container.firstChild).toMatchSnapshot()
    })
  })
  describe('handling input', () => {
    it('should set a value when passed in', () => {
      const { container } = render(<Input initialValue="My Input Value" />)

      expect(container.firstChild).toMatchSnapshot()
      const input = container.firstChild
      expect(input.value).toBe('My Input Value')
    })
    it('should update a value when changed', () => {
      const { container } = render(<Input initialValue="My Input Value" />)
      const input = container.firstChild

      fireEvent.change(input, { target: { value: 'Updated Value' } })
      expect(input.value).toBe('Updated Value')
      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
