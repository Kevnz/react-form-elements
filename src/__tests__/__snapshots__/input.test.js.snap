// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component The Rendered HTML should have a placeholder when passed 1`] = `
<input
  id="id-2-1"
  name="Input"
  placeholder="textbox placeholder"
  type="text"
  value=""
/>
`;

exports[`Input component The Rendered HTML should render input element 1`] = `
<input
  id="id-1-1"
  name="Input"
  type="text"
  value=""
/>
`;

exports[`Input component handling input should set a value when passed in 1`] = `
<input
  id="id-3-1"
  name="Input"
  type="text"
  value="My Input Value"
/>
`;

exports[`Input component handling input should update a value when changed 1`] = `
<input
  id="id-5-1"
  name="Input"
  type="text"
  value="Updated Value"
/>
`;
